### PR TITLE
Added RPCO bits to test-upgrade and..

### PR DESCRIPTION
**Pull request by alextricity25**
Also moved the rpc playbook bits from deploy.sh into
it's own script, deploy-rpc-playbooks.sh. This is so
those bits can be easily included in test-upgrades.sh and in
deploy.sh.
I moved the run_ansible function previously defined in deploy.sh to a new file, ``functions.sh``, that is
then sourced by both ``deploy.sh`` and ``test-upgrades.sh``.
The bash environment variables is another matter. For now, I just exported what seemed necessary into the environment so the variables can be carried over to the next script. However, we should look into creating a more streamlined way to manage these variables later. Perhaps move the variables into their own file, that other scripts can then source.

Connects https://github.com/rcbops/u-suk-dev/issues/393

### CODE REVIEW

**Comment by rackernexus** [comment]
HULK [PREVIEW](https://staging.developer.rackspace.com/staging.horse/build-f3dd8ff5bd/docs/private-cloud/release-notes/)

**Comment by rackernexus** [comment]
If you want to double-check your formatting, check [the preview](https://staging.developer.rackspace.com/staging.horse/build-68e18c81c7/docs/private-cloud/release-notes/).

**Comment by rackernexus** [comment]
:eyes: [Preview](https://staging.developer.rackspace.com/staging.horse/build-02b1aa6a9e/docs/private-cloud/release-notes/)

**Comment by alextricity25** [comment]
This code is ready for review, but I want to wait for https://github.com/rcbops/rpc-openstack/pull/1395 to merge so I can deal with the merge conflicts before they come back to haunt me.

**Comment by rackernexus** [comment]
HULK [PREVIEW](https://staging.developer.rackspace.com/staging.horse/build-6a5daa3da0/docs/private-cloud/release-notes/)

**Comment by rackernexus** [comment]
HULK [PREVIEW](https://staging.developer.rackspace.com/staging.horse/build-e2e8d99c7b/docs/private-cloud/release-notes/)

File: scripts/test-upgrade.sh [hunk 1]
```diff
@@ -22,30 +22,54 @@ BASE_DIR=$( cd "$( dirname ${0} )" && cd ../ && pwd )
...
+    --defaults "${RPCD_DIR}"/etc/openstack_deploy/user_osa_variables_defaults.yml \
+    --overrides /etc/openstack_deploy/user_variables.yml \
+    --output-file /etc/openstack_deploy/user_osa_variables_overrides.yml
+fi
+# TODO: We need to eventually find a proper way to migrate these secret variables.
+#       For now, we just copy existing secret variables defined previously on the
+#       liberty install. This won't cause any problems since there aren't any new
+#       secret variables in mitaka that are not defined in liberty.
+cp /etc/openstack_deploy/user_extras_secrets.yml \
+   /etc/openstack_deploy/user_rpco_secrets.yml
```
Comment (mattt416) [inline scripts/test-upgrade.sh]: Do you skip moving user_secrets.yml to user_osa_secrets.yml since that's handled by deploy.sh?

File: scripts/test-upgrade.sh [hunk 2]
```diff
@@ -55,16 +79,25 @@ cd ${OA_DIR}/playbooks
...
-# TASK #4 TODO: This task should be moved somewhere towards the bottom once
-#               we get a more robust script going.
+# TASK #5
+# https://github.com/rcbops/u-suk-dev/issues/393
+# Run deploy.sh with MaaS and don't run the OSA playbooks.
+# Ultimitely, this will run the RPCO playbooks.
+cd ${BASE_DIR}
+export DEPLOY_MAAS=${DEPLOY_MAAS:-'yes'}
+export DEPLOY_OA='no'
+export BOOTSTRAP_ANSIBLE='no'
```
Comment (mattt416) [inline scripts/test-upgrade.sh]: deploy.sh doesn't use BOOTSTRAP_ANSIBLE at the moment, is that a work in progress?

**Comment by alextricity25** [comment]
This is ready for review.
For those who are testing: Keep in mind that unless you pull in[ this patch](https://review.openstack.org/#/c/360539/), or wait until the next SHA bump, the post-upgrade playbooks is expected to fail.

File: scripts/test-upgrade.sh (hunk 2 above)
Comment (alextricity25) [inline scripts/test-upgrade.sh]: I added it in as part of this PR. See changes in deploy.sh

File: scripts/test-upgrade.sh (hunk 1 above)
Comment (alextricity25) [inline scripts/test-upgrade.sh]: That's correct!

File: scripts/test-upgrade.sh (hunk 2 above)
Comment (mattt416) [inline scripts/test-upgrade.sh]: *goes to get more coffee*

**Comment by mattt416** [comment]
Testing this now on a liberty deploy.

**Comment by mattt416** [comment]
My upgrade has progressed to the rpc-openstack bits, so this LGTM 👍

**Comment by rackernexus** [comment]
:mag_right: [Preview](https://staging.developer.rackspace.com/staging.horse/build-f8e5d924df/docs/private-cloud/release-notes/)

File: scripts/test-upgrade.sh [hunk 3]
```diff
@@ -22,30 +22,54 @@ BASE_DIR=$( cd "$( dirname ${0} )" && cd ../ && pwd )
 OA_DIR="$BASE_DIR/openstack-ansible"
 RPCD_DIR="$BASE_DIR/rpcd"
 
+# Create a backup of /etc/openstack_deploy, if one doesn't already exist
+# NOTE: This gets done by openstack-ansible's scripts/run-upgrade.sh script,
+#       but if we wait until then it'll be too late.
+if [[ ! -d "/etc/openstack_deploy.MITAKA" ]]; then
+  cp -a /etc/openstack_deploy /etc/openstack_deploy.MITAKA
```
Comment (sigmavirus24) [inline scripts/test-upgrade.sh]: if this is the backup, should it be called "MITAKA"? It's a back up of Liberty's deploy directory, right?

Comment (evrardjp) [inline scripts/test-upgrade.sh]: That's how we do in OSA. It's fine for me if we say it's because it's happening during mitaka upgrade that we do that :)

File: scripts/test-upgrade.sh [hunk 4]
```diff
@@ -22,30 +22,54 @@ BASE_DIR=$( cd "$( dirname ${0} )" && cd ../ && pwd )
...
-rm /etc/openstack_deploy/user_extras_variables.yml /etc/openstack_deploy/user_variables.yml
+rm /etc/openstack_deploy/user_extras_variables.yml \
+   /etc/openstack_deploy/user_variables.yml        \
+   /etc/openstack_deploy/user_extras_secrets.yml
 
 # TASK #1
-# Bug: https://github.com/rcbops/u-suk-dev/issues/293
+# Update Ansible
+cd ${OA_DIR}
+bash scripts/bootstrap-ansible.sh
```
Comment (evrardjp) [inline scripts/test-upgrade.sh]: Why do we need to bootstrap-ansible again? Would it be possible to use the first bootstrap ansible to install ansible and the roles, and here just refresh the roles (only if needed)?

File: scripts/test-upgrade.sh (hunk 2 above)
Comment (evrardjp) [inline scripts/test-upgrade.sh]: good to remove that part.

File: scripts/test-upgrade.sh (hunk 4 above)
Comment (alextricity25) [inline scripts/test-upgrade.sh]: I honestly don't remember why I needed bootstrap ansible here.
What did you have in mind? ``ansible-galaxy --install`` to update the roles?

Comment (andymcc) [inline scripts/test-upgrade.sh]: You need the openstack_hosts role in order to run the setup_hosts role in Task #3
The other option is looking at whether we need to rebuild the containers here... OSA upgrade will do a container-create (except for galera_all) so it should recreate them, the only issue would be if the repo servers are needed before that happens.

Comment (alextricity25) [inline scripts/test-upgrade.sh]: @andymcc Good catch on ``pip.conf``. I'll add that task in.
RE: repo-containers, The reason we needed them up before the OSA upgrade script is because that script runs [a playbook](https://github.com/openstack/openstack-ansible/blob/stable/mitaka/scripts/run-upgrade.sh#L152) that targets repo-containers before running ``setup-hosts.yml``. And if they are not there before it's run, it will fail :(

Comment (andymcc) [inline scripts/test-upgrade.sh]: Ahh ok cool - I was going to test that, makes sense - would've been nicer the other way but I did the same thing to get it going earlier, so lets roll with that.

File: scripts/test-upgrade.sh (hunk 3 above)
Comment (sigmavirus24) [inline scripts/test-upgrade.sh]: > That's how we do in OSA.

I'm fairly certain that RPCO has done it differently in the past though. It also makes more sense to me to label the directory with the release for which it works.

@d34dh0r53 and @Apsu you two seemed to be confused by this too on Thursday and Friday. Thoughts?

File: scripts/test-upgrade.sh (hunk 1 above)
Comment (andymcc) [inline scripts/test-upgrade.sh]: Can we add:
if [[ -f "/etc/openstack_deploy/user_extra_secrets.yml" ]]; then
  cp ...
  cp ...
  rm ...
fi

**Comment by rackernexus** [comment]
HULK [PREVIEW](https://staging.developer.rackspace.com/staging.horse/build-0926a96e24/docs/private-cloud/release-notes/)

**Comment by alextricity25** [comment]
Ready for testing

**Comment by rackernexus** [comment]
HULK [PREVIEW](https://staging.developer.rackspace.com/staging.horse/build-8598d8dd25/docs/private-cloud/release-notes/)

File: scripts/test-upgrade.sh (hunk 3 above)
Comment (andymcc) [inline scripts/test-upgrade.sh]: I agree with @sigmavirus24 it makes more sense (to me at least) for it to be openstack_deploy.LIBERTY - since that would be the version for which the variables are setup, its a simple change too.

File: scripts/test-upgrade.sh [hunk 5]
```diff
@@ -22,30 +22,57 @@ BASE_DIR=$( cd "$( dirname ${0} )" && cd ../ && pwd )
...
+  cp -a ${RPCD_DIR}/etc/openstack_deploy/*defaults* /etc/openstack_deploy
+  rm /etc/openstack_deploy/user_extras_variables.yml \
+     /etc/openstack_deploy/user_variables.yml        \
+     /etc/openstack_deploy/user_extras_secrets.yml
+fi
 
 # TASK #1
-# Bug: https://github.com/rcbops/u-suk-dev/issues/293
+# Update Ansible
+rm /root/.pip/pip.conf
```
Comment (andymcc) [inline scripts/test-upgrade.sh]: Should we do an if on this, or a || true so that the rm will succeed even if the file doesn't exist.

Comment (alextricity25) [inline scripts/test-upgrade.sh]: Good point. I'll add that now

**Comment by rackernexus** [comment]
:eyes: [Preview](https://staging.developer.rackspace.com/staging.horse/build-eb0358d94b/docs/private-cloud/release-notes/)

**Comment by rackernexus** [comment]
If you want to double-check your formatting, check [the preview](https://staging.developer.rackspace.com/staging.horse/build-8760b84c90/docs/private-cloud/release-notes/).

**Comment by git-harry** [comment]
👎 this looks to be adding a lot of complexity. I think we should consider breaking out bits of deploy.sh so that both deploy.sh and test-upgrade.sh are using the same scripts instead of trying to make deploy.sh have more configurability. So for example below is the deploy.sh code to run the RPC playbooks, why not break that out into a separate script and have both deploy.sh and test-upgrade.sh use it. I think that would make it clearer what is shared between the two scripts, the current deploy.sh has already proved problematic for support to understand, this patch seems to be increasing the complexity.

```
# begin the RPC installation
cd ${RPCD_DIR}/playbooks/

# configure everything for RPC support access
run_ansible rpc-support.yml

# configure the horizon extensions
run_ansible horizon_extensions.yml

# deploy and configure RAX MaaS
if [[ "${DEPLOY_MAAS}" == "yes" ]]; then
  run_ansible setup-maas.yml
  run_ansible verify-maas.yml
fi

# deploy and configure the ELK stack
if [[ "${DEPLOY_ELK}" == "yes" ]]; then
  run_ansible setup-logging.yml

  # deploy the LB required for the ELK stack
  if [[ "${DEPLOY_HAPROXY}" == "yes" ]]; then
    run_ansible haproxy.yml
  fi  
fi
```

File: scripts/test-upgrade.sh [hunk 6]
```diff
@@ -22,30 +22,59 @@ BASE_DIR=$( cd "$( dirname ${0} )" && cd ../ && pwd )
...
+fi
 
 # TASK #1
-# Bug: https://github.com/rcbops/u-suk-dev/issues/293
+# Update Ansible
+if [[ -f "/root/.pip/pip.conf" ]]; then
+  rm /root/.pip/pip.conf
+fi
+cd ${OA_DIR}
+bash scripts/bootstrap-ansible.sh
```
Comment (evrardjp) [inline scripts/test-upgrade.sh]: why are we rebootstrapping ansible?

Shouldn't we just do one bootstrap once (install ansible and roles), and skip the multiple bootstrap?

**Comment by evrardjp** [comment]
Agreed with @git-harry 
let's split in small bits, and re-use just what we need if possible.

File: scripts/test-upgrade.sh (hunk 6 above)
Comment (alextricity25) [inline scripts/test-upgrade.sh]: We need the roles to be in place before we run ``setup-hosts.yml`` in TASK#3.
I'm not sure what you mean by "rebootstrapping ansible". This is the first invocation of ``bootstrap-ansible.sh`` in the script, no?

Comment (Apsu) [inline scripts/test-upgrade.sh]: Line 38 does as well, if the variable is set.

**Comment by alextricity25** [comment]
I've incorporated the feedback above. Once I finish testing it out I'll submit the latest commit.

**Comment by rackernexus** [comment]
Your content preview [is now ready](https://staging.developer.rackspace.com/staging.horse/build-89a75b2153/docs/private-cloud/release-notes/). :bow:

**Comment by rackernexus** [comment]
HULK [PREVIEW](https://staging.developer.rackspace.com/staging.horse/build-633d023962/docs/private-cloud/release-notes/)

**Comment by alextricity25** [comment]
recheck_swift

**Comment by rackernexus** [comment]
:mag_right: [Preview](https://staging.developer.rackspace.com/staging.horse/build-18c3592dc6/docs/private-cloud/release-notes/)

**Comment by rackernexus** [comment]
HULK [PREVIEW](https://staging.developer.rackspace.com/staging.horse/build-15ab25748b/docs/private-cloud/release-notes/)

File: scripts/deploy-rpc-playbooks.sh [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+# deploy and configure the ELK stack
+if [[ "${DEPLOY_ELK}" == "yes" ]]; then
+  run_ansible setup-logging.yml
+
+  # deploy the LB required for the ELK stack
+  if [[ "${DEPLOY_HAPROXY}" == "yes" ]]; then
+    run_ansible haproxy.yml
+  fi
+fi
```
Comment (mattt416) [inline scripts/deploy-rpc-playbooks.sh]: Should this script be able to be run independently?  I see benefit in that, but then it means having another place where default variables need to be maintained (currently 3 places).

**Comment by alextricity25** [comment]
@all Please review the latest commit. Thank you!

**Comment by rackernexus** [comment]
HULK [PREVIEW](https://staging.developer.rackspace.com/staging.horse/build-2c76369a0a/docs/private-cloud/release-notes/)

File: scripts/test-upgrade.sh [hunk 8]
```diff
@@ -18,34 +18,65 @@
...
+export BASE_DIR=$( cd "$( dirname ${0} )" && cd ../ && pwd )
+export OA_DIR="$BASE_DIR/openstack-ansible"
+export RPCD_DIR="$BASE_DIR/rpcd"
+
+source ${BASE_DIR}/scripts/functions.sh
+
+# Create a backup of /etc/openstack_deploy, if one doesn't already exist
+# NOTE: This gets done by openstack-ansible's scripts/run-upgrade.sh script,
+#       but if we wait until then it'll be too late.
+if [[ ! -d "/etc/openstack_deploy.MITAKA" ]]; then
```
Comment (mattt416) [inline scripts/test-upgrade.sh]: Looks like this is still referencing the old `MITAKA`.  Side note, I created https://review.openstack.org/#/c/369650/ so that upstream OSA upgrade backs stuff up to /etc/openstack_deploy.LIBERTY, rather than /etc/openstack_deploy.MITAKA.  In liberty of OSA it backed up to /etc/openstack_deploy.KILO.

With that said, I almost feel like this script should back up somewhere else, ie. /etc/openstack_deploy.LIBERTY_RPCO or whatever.  The reason being that when OSA upgrade commences and backs up these files it's going to overwrite the original backups we had in /etc/openstack_deploy.LIBERTY.  It would be good to have a pristine backup of /etc/openstack_deploy which we know OSA won't touch that support can resort to if needed.  Thoughts?

File: scripts/test-upgrade.sh [hunk 9]
```diff
@@ -18,34 +18,65 @@
...
-    --overrides /etc/openstack_deploy/user_extras_variables.yml \
-    --output-file /etc/openstack_deploy/user_rpco_variables_overrides.yml
-"${BASE_DIR}"/scripts/migrate-yaml.py --for-testing-take-new-vars-only \
-  --defaults "${RPCD_DIR}"/etc/openstack_deploy/user_osa_variables_defaults.yml \
-  --overrides /etc/openstack_deploy/user_variables.yml \
-  --output-file /etc/openstack_deploy/user_osa_variables_overrides.yml
-cp -a ${RPCD_DIR}/etc/openstack_deploy/*defaults* /etc/openstack_deploy
-rm /etc/openstack_deploy/user_extras_variables.yml /etc/openstack_deploy/user_variables.yml
+if [[ ! -f "/etc/openstack_deploy/user_rpco_variables_overrides.yml" ]]; then
+  "${BASE_DIR}"/scripts/migrate-yaml.py --for-testing-take-new-vars-only \
```
Comment (mattt416) [inline scripts/test-upgrade.sh]: It'd probably be worthwhile pointing out that `--for-testing-take-new-vars-only` should in no way be used in production.

File: scripts/test-upgrade.sh (hunk 8 above)
Comment (alextricity25) [inline scripts/test-upgrade.sh]: Good thinking. This is actually [already done in the pre-upgrade playbooks](https://github.com/rcbops/rpc-openstack/blob/master/rpcd/playbooks/roles/rpc_pre_upgrade/tasks/backup.yml), so it can be removed.

**Comment by rackernexus** [comment]
HULK [PREVIEW](https://staging.developer.rackspace.com/staging.horse/build-b416349db4/docs/private-cloud/release-notes/)

**Comment by alextricity25** [comment]
I was tempted to move all of the bash environment variables into one file that we could source(that way we have one place for all of these variables instead of three), but I'm not so sure if that work should be part of this card. 
@mattt416 I've addressed your comments.
The latest commit is ready for review. Thanks.

**Comment by rackernexus** [comment]
:mag_right: [Preview](https://staging.developer.rackspace.com/staging.horse/build-a368506ccb/docs/private-cloud/release-notes/)

**Comment by rackernexus** [comment]
:mag_right: [Preview](https://staging.developer.rackspace.com/staging.horse/build-d09d31705f/docs/private-cloud/release-notes/)

**Comment by rackernexus** [comment]
If you want to double-check your formatting, check [the preview](https://staging.developer.rackspace.com/staging.horse/build-ea4cdff1ab/docs/private-cloud/release-notes/).

File: scripts/deploy-rpc-playbooks.sh (hunk 7 above)
Comment (git-harry) [inline scripts/deploy-rpc-playbooks.sh]: Yes, this script has to be able to be run separately. The whole idea is that each step in test-upgrade.sh is either something support needs to be able to run or test 'glue'.

File: scripts/test-upgrade.sh [hunk 10]
```diff
@@ -18,34 +18,59 @@
...
+     /etc/openstack_deploy/user_rpco_secrets.yml
+fi
 cp -a ${RPCD_DIR}/etc/openstack_deploy/*defaults* /etc/openstack_deploy
-rm /etc/openstack_deploy/user_extras_variables.yml /etc/openstack_deploy/user_variables.yml
 
 # TASK #1
-# Bug: https://github.com/rcbops/u-suk-dev/issues/293
+# Update Ansible
+if [[ -f "/root/.pip/pip.conf" ]]; then
+  rm /root/.pip/pip.conf
```
Comment (git-harry) [inline scripts/test-upgrade.sh]: Why is this required? bootstrap-ansible.sh falls back on --isolated where necessary. Did you see an error?

**Comment by andymcc** [comment]
We still need to setup the ceph playbooks, its possible the directory is moved and a new one cloned - so the roles won't exist - causing MaaS to fail straight away.
e.g. these lines: https://github.com/rcbops/rpc-openstack/blob/master/scripts/deploy.sh#L37-L41

**Comment by git-harry** [comment]
@andymcc I noted that on https://github.com/rcbops/u-suk-dev/issues/388 but I didn't realise MaaS was dependent on them. Maybe we should have a bootstrap script that does bootstrap-ansible.sh and these RPCO bits together.

**Comment by andymcc** [comment]
Ok with the ceph change - this worked for me - there is one bug in the rpc_post_upgrade role, which will require a backport in OSA (or a work around - which may be better for now in RPC) - will create a card for that.

I'm happy to merge this once the Ceph bit is fixed.

File: scripts/test-upgrade.sh (hunk 10 above)
Comment (andymcc) [inline scripts/test-upgrade.sh]: It spits out errors that can be confusing - although it will work.

**Comment by rackernexus** [comment]
Your content preview [is now ready](https://staging.developer.rackspace.com/staging.horse/build-e7b2a36bec/docs/private-cloud/release-notes/). :bow:

File: scripts/test-upgrade.sh [hunk 11]
```diff
@@ -71,16 +101,25 @@ cd ${OA_DIR}/playbooks
...
+# https://github.com/rcbops/u-suk-dev/issues/393
+# Run deploy-rpc-playbooks.sh
+# Ultimitely, this will run the RPCO playbooks.
+cd ${BASE_DIR}
+export DEPLOY_HAPROXY=${DEPLOY_HAPROXY:-"no"}
+export DEPLOY_ELK=${DEPLOY_ELK:-"yes"}
+export DEPLOY_MAAS=${DEPLOY_MAAS:-"yes"}
+bash scripts/deploy-rpc-playbooks.sh
+
+# TASK #7
```
Comment (git-harry) [inline scripts/test-upgrade.sh]: 8

File: scripts/test-upgrade.sh (hunk 6 above)
Comment (andymcc) [inline scripts/test-upgrade.sh]: I believe thats in the deploy.sh script which doesn't actually get called by test-upgrade.sh
scripts/deploy-rpc-playbooks.sh does get called, but that doesn't perform a bootstrap.

We do end up with 2 bootstraps unfortunately, because one happens as part of the openstack-ansible run-upgrade.sh, but that seems to be unavoidable if we want to remove and recreate the repo servers (Task 4). Or at least we do need to take some actions and performing a bootstrap just seems simpler than (for example) installing a subset of roles and hoping all dependencies etc are met.

**Comment by git-harry** [comment]
👎 scripts/deploy-rpc-playbooks.sh doesn't look like something that support could easily run, they have to set all sorts of variables just to run it.

File: scripts/test-upgrade.sh (hunk 10 above)
Comment (evrardjp) [inline scripts/test-upgrade.sh]: Agreed it's just on the deploy node anyway.

File: scripts/test-upgrade.sh (hunk 6 above)
Comment (evrardjp) [inline scripts/test-upgrade.sh]: Well, now that we broke the test-upgrade in multiple script this occurence is indeed the first one.

Because we run the osa bootsrap and then install the rpc roles through galaxy I think we are fine.

Yes, we re-bootstrap during the upgrade, but it seems a pain to change in OSA, not RPC.

So, we're good.

**Comment by alextricity25** [comment]
@git-harry I disagree that ``deploy-rpc-playbooks.sh`` should be something that can be independently run. It should only be invoked by either ``deploy.sh`` or ``test-upgrades.sh``. If someone wants to run the rpc-playbooks, they can just ``cd /opt/rpc-openstack/rpcd/playbooks`` and run them from there(which is what I hope support would be doing when upgrading an environment that's not for testing)
I don't agree with setting defaults for these variables in various places that we'll have to keep track of. i.e.
```
export ${DEPLOY_MAAS:-'yes'}
..
..
```

I think the right way to go would be to streamline the way all of these variables are set. We could munge them into one file, then have these deployment/upgrade scripts source that file. I don't believe that's part of the scope of this card, though.

**Comment by andymcc** [comment]
👍  This worked for me in a full multi-node upgrade.

**Comment by mattt416** [comment]
I'm going to test this out today, but generally this looks OK to me.  We'll still need to figure out where the ceph playbooks are re-run but we can add that as we start digging into the ceph bits.  Only final thing that I can think of that is needed (now that we do not include deploy.sh) is moving user_secrets.yml to user_osa_secrets.yml.

**Comment by alextricity25** [comment]
recheck_ceph

**Comment by mattt416** [comment]
Tested, had some unrelated failures along the way (ssh timeouts etc.) but this script otherwise looks 👍 to me.

**Comment by alextricity25** [comment]
recheck_ceph

**Comment by alextricity25** [comment]
recheck_ceph

**Comment by git-harry** [comment]
recheck_ceph

**Comment by hughsaunders** [comment]
recheck_ceph

**Comment by git-harry** [comment]
recheck_ceph

**Comment by git-harry** [comment]
recheck_ceph

**Comment by evrardjp** [comment]
recheck_ceph -- maas issue.

**Comment by mancdaz** [comment]
recheck_ceph

**Comment by alextricity25** [comment]
recheck_ceph

**Comment by alextricity25** [comment]
recheck_ceph

**Comment by alextricity25** [comment]
recheck_ceph

**Comment by mancdaz** [comment]
recheck_ceph

**Comment by alextricity25** [comment]
recheck_all

**Comment by Apsu** [comment]
recheck_ceph

**Comment by mancdaz** [comment]
recheck_swift

**Comment by mattt416** [comment]
recheck_ceph

**Comment by alextricity25** [comment]
recheck_ceph !!!!!

**Comment by mancdaz** [comment]
w00t!!
